File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/RunConfigDto.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class RunConfigDto
    {
        [JsonProperty("runId")]
        public Guid runId;

        [JsonProperty("hero")]
        public RunHeroProgressDto hero;

        [JsonProperty("encounters")]
        public List<RunEncounterDto> encounters = new List<RunEncounterDto>();

        [JsonProperty("learnedMoves")]
        public List<MoveDto> learnedMoves = new List<MoveDto>();

        [JsonProperty("equippedMoveIds")]
        public List<long> equippedMoveIds = new List<long>();

        [JsonProperty("completed")]
        public bool completed;
    }
}

File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/RunConfigDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RunConfigDTO {

    private UUID runId;
    private RunHeroProgressDTO hero;
    private List<RunEncounterDTO> encounters;
    private List<MoveDTO> learnedMoves;
    private List<Long> equippedMoveIds;
    private boolean completed;
}